### PR TITLE
[Opmondev-165]:  anonymize Opendata

### PR DESCRIPTION
**Pull request by wisecrow**
Anonymize Opendata.

This is done by adding  optional parameter`--only_opendata` to anonymizer executable.

### CODE REVIEW

File: anonymizer_module/opmon_anonymizer/iio/mongodbmanager.py [hunk 1]
```diff
@@ -40,17 +44,22 @@ def __init__(self, settings, logger):
...
         user = settings['mongodb']['user']
         password = urllib.parse.quote(settings['mongodb']['password'], safe='')
         host = settings['mongodb']['host']
-        return f"mongodb://{user}:{password}@{host}/auth_db"
+        return f'mongodb://{user}:{password}@{host}/auth_db'
+
+
+class MongoDbManager(BaseMongoDbManager):
+
+    def __init__(self, settings, logger):
```
Comment (justinask7) [inline anonymizer_module/opmon_anonymizer/iio/mongodbmanager.py:60]: Type hints would be nice

File: anonymizer_module/opmon_anonymizer/opendata_anonymizer.py [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+                last_successful_batch_timestamp = batch_end_mongodb_timestamp
+
+        if record_buffer:
+            self._anonymization_job.run(record_buffer)
+            processed_count += len(record_buffer)
+
+        return processed_count
+
+    @staticmethod
+    def _get_allowed_fields(field_translations_file_path: str, logger: LoggerManager) -> list:
```
Comment (justinask7) [inline anonymizer_module/opmon_anonymizer/opendata_anonymizer.py:113]: Hmm, logger must be passed all the time? 🤔 looks strange… it should be global IMHO or at least given to instance on init time

**Review by justinask7 — APPROVED** [review]

File: anonymizer_module/opmon_anonymizer/opendata_anonymizer.py (hunk 2 above)
Comment (wisecrow) [reply anonymizer_module/opmon_anonymizer/opendata_anonymizer.py:113]: This is done due to `_get_allowed_fields()` being static method, which cannot access object attribute `self._logger`.
But it expected still to have all setup. I would leave it here for now.

File: anonymizer_module/opmon_anonymizer/main.py [hunk 3]
```diff
@@ -19,40 +19,56 @@
...
+            logger.log_error('opendata_anonymization_process_failed', f'Failed to anonymize. ERROR: {trace}')
+            raise
+    else:
+        try:
+            start_time = datetime.now()
+            record_count = run_anonymizer(settings, logger, args.limit)
+            elapsed = str(datetime.now() - start_time)
+            logger.log_info('anonymization_session_finished',
+                            f'Anonymization finished in {elapsed} seconds. Processed {record_count} entries from MongoDB.')
+        except Exception:
```
[the remark this answers is not in the document]
Comment (wisecrow) [reply anonymizer_module/opmon_anonymizer/main.py]: Refactored to a function. Thanks for comment.

**Pull request merged**